sdlib/d/gc/sizeclass.d

```diff
@@ -47,6 +47,69 @@ enum SizeClass {
 
 enum MaxTinySize = ClassCount.Tiny * Quantum;
 
+// Determine whether given size is permitted by allocator.
+bool isAllocatableSize(size_t size) {
+	return size > 0 && size <= MaxAllocationSize;
+}
+
+unittest isAllocatableSize {
+	assert(!isAllocatableSize(0));
+	assert(isAllocatableSize(1));
+	assert(isAllocatableSize(42));
+	assert(isAllocatableSize(99999));
+	assert(isAllocatableSize(MaxAllocationSize));
+	assert(!isAllocatableSize(MaxAllocationSize + 1));
+	assert(!isAllocatableSize(size_t.max));
+}
+
+enum maxSizeClass = ClassCount.Total - 1;
+
+// Determine whether given size class is permitted by the allocator.
+bool isAllocatableSizeClass(uint sizeClass) {
+	return sizeClass <= maxSizeClass;
+}
+
+// Determine whether given size class is considered 'small' (slab-allocatable).
+bool isSmallSizeClass(uint sizeClass) {
+	return sizeClass < ClassCount.Small;
+}
+
+// Determine whether given size may fit into a 'small' (slab-allocatable) size class.
+bool isSmallSize(size_t size) {
+	return isAllocatableSize(size) && (size <= SizeClass.Small);
+}
+
+// Determine whether given size class is considered 'large'.
+bool isLargeSizeClass(uint sizeClass) {
+	assert(isAllocatableSizeClass(sizeClass));
+	return sizeClass >= ClassCount.Small;
+}
+
+// Determine whether given size may fit into a 'large' size class.
+bool isLargeSize(size_t size) {
+	return isAllocatableSize(size) && (size > SizeClass.Small);
```

Comment:
Why write it in such a way when 2 compares are fine? The fact large size needs to be relocatable is a property we want to test anyways.